api/indexing_router.py

```
@@ -59,20 +60,25 @@ def index(request: IndexRequest, req: Request):
             raise HTTPException(status_code=500, detail="EMBEDDING_DIMENSION_MISMATCH")
-        # Upsert
+        # Create index if needed, then upsert vectors
         try:
-            vdb.upsert(namespace=request.project_id, ids=ids, vectors=vectors, metadata=metadata)
+            # Ensure index is initialized first
+            vdb.create_index()
+            # Then upsert vectors with the proper API
+            vdb.upsert_vectors(vectors=vectors, ids=ids)
+            # Note: metadata support will be added in a follow-up
         except ValueError as e:
             raise HTTPException(status_code=422, detail=str(e))
```

Ruff (0.13.1)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)